### PR TITLE
react_carousel

### PR DESCRIPTION
**Pull request by georgiyevv**
[DEMO LINK](https://svyatoslavPy.github.io/react_carousel/)

### CODE REVIEW

**Review by sTorba24 — CHANGES_REQUESTED** [review]
Good work!  But you need some improvements
Try to pass all tests before requesting review

If you have any issues with it - feel free to text your fe_chat

These settings should be changeable, use inputs etc
image

**Review by ghost — CHANGES_REQUESTED** [review]
No changes are deployed

File: src/App.tsx [hunk 1]
```diff
@@ -20,17 +20,38 @@ class App extends React.Component<{}, State> {
       './img/9.png',
       './img/10.png',
     ],
+    step: 1,
```
Comment (sTorba24) [inline src/App.tsx:23]: 3 by default

**Review by sTorba24 — CHANGES_REQUESTED** [review]
Good progress!

1. If there not enough items to scroll for full step count => scroll only count of images that available. Here shouldn't be 3 white spaces
image

2. disable prev/next when you already scrolled to first/last item
image

File: src/components/Carousel/Carousel.scss [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    // column-gap: 10px;
+    width: 100%;
+    margin: 0;
+    padding: 0;
+  }
+
+  // &__picture {
+  //   width: 130px;
+  //   height: 130px;
+  // }
```
Comment (volodymyr-soltys97) [inline src/components/Carousel/Carousel.scss:28]: Remove comments everywhere

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Good job 👍 
Let's improve your code
Need to fix width
image

Also need to remove list style for the tag li
image

File: src/App.tsx [hunk 3]
```diff
@@ -20,17 +27,38 @@ class App extends React.Component<{}, State> {
...
 
   render() {
-    const { images } = this.state;
+    const { images, step, frameSize, itemWidth, animationDuration } =
+      this.state;
 
     return (
       <div className="App">
         {/* eslint-disable-next-line */}
         <h1>Carousel with {images.length} images</h1>
```
Comment (anastasiiavorobiova) [inline src/App.tsx:52]:
```suggestion
        <h1 data-cy="title">Carousel with {images.length} images</h1>
```
Consider adding all needed data attributes to pass tests

File: src/components/CarouselSettings/CarouselSettings.scss [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+$gray-color: rgb(235, 235, 235);
+
+.Carousel-settings {
+  width: 350px;
+  height: 380px;
+  padding: 41px 55px 8px 39px;
+  box-sizing: border-box;
+  border: 1px solid $gray-color;
+  border-radius: 12px;
+  box-shadow: 0px 2.31px 8.67px 0px $white-color;
```
Comment (anastasiiavorobiova) [inline src/components/CarouselSettings/CarouselSettings.scss:11]:
```suggestion
  box-shadow: 0 2.31px 8.67px 0 $white-color;
```

File: src/components/CarouselSettings/CarouselSettings.scss [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  background: $white-color;
+  font-weight: 400;
+  font-size: 17px;
+  transition: all 0.5s;
+  cursor: pointer;
+
+  &:hover {
+    box-shadow: 0px 2.31px 8.67px 0px $gray-color;
```
Comment (anastasiiavorobiova) [inline src/components/CarouselSettings/CarouselSettings.scss:19]:
```suggestion
    box-shadow: 0 2.31px 8.67px 0 $gray-color;
```

File: src/components/Carousel/Carousel.tsx [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+            {images.map((image: string) => (
+              <li key={image} className="Carousel__item">
+                <img
+                  style={{
+                    width: `${itemWidth}px`,
+                    height: `${itemWidth}px`,
+                  }}
+                  className="Carousel__picture"
+                  src={image}
+                  alt="1"
```
Comment (anastasiiavorobiova) [inline src/components/Carousel/Carousel.tsx:76]:
```suggestion
                  alt="1"
                  width={itemWidth}
```

File: src/components/Carousel/Carousel.tsx [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+              transform: `translateX(-${ContentWidth}px)`,
+              transition: `transform ${animationDuration}ms`,
+            }}
+            className="Carousel__list"
+          >
+            {images.map((image: string) => (
+              <li key={image} className="Carousel__item">
+                <img
+                  style={{
+                    width: `${itemWidth}px`,
```
Comment (anastasiiavorobiova) [inline src/components/Carousel/Carousel.tsx:71]:
```suggestion

```

File: src/components/Carousel/Carousel.scss
Comment (anastasiiavorobiova) [inline src/components/Carousel/Carousel.scss:1]: Consider adding space between carousel images

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Great work! Almost done!

**Pull request closed without merging**